src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -601,27 +603,40 @@ contract ConvexAvatarMultiToken is BaseAvatar, ConvexAvatarUtils, PausableUpgrad
     /// @param _lpAmount Amount of bpt to be withdrawn
     function _withdrawLpToUnderlyings(address _lpToken, uint256 _lpAmount) internal {
         ICurvePool pool = ICurvePool(META_REGISTRY.get_pool_from_lp_token(_lpToken));
+        if (address(pool) == address(0)) revert PoolNotFoundInMetaregistry(_lpToken);
         uint256 coins = META_REGISTRY.get_n_coins(address(pool));
 
         if (coins == TWO_COINS_POOL) {
             uint256[2] memory minOutAmounts;
             // @audit DAO is accepting under emergency situations for speed defaulting
             // the amount out expected to zero in this ocassion
             pool.remove_liquidity(_lpAmount, minOutAmounts);
-        } else {
+        } else if (coins == THREE_COINS_POOL) {
             // NOTE: pools may have three tokens like tricrypto2/3pool needs
             // different signature to withdraw
             uint256[3] memory minOutAmounts;
-            // @audit idem comment as L598-L599
+            // @audit idem comment as L611-612
+            pool.remove_liquidity(_lpAmount, minOutAmounts);
+        } else {
+            // NOTE: pools may have four tokens like ySwap needs
+            // different signature to withdraw
+            uint256[4] memory minOutAmounts;
+            // @audit idem comment as L611-612
             pool.remove_liquidity(_lpAmount, minOutAmounts);
         }
```

Comment:
again, this should be explicit.

i think the else should only capture cases that cant be predicted in advance, so `coins` isnt 2, 3 or 4.

Reply:
similar to the discussion here: https://github.com/Badger-Finance/badger-safe-modules/pull/7#discussion_r1097175962

Reply:
explicity else if added in dbdb8a3656039a6fbe557e5d3e4fca8ec2b2d091